### PR TITLE
Feature/ADF-213/Asset Management - Atomic roles - batch 5

### PR DESCRIPTION
**Pull request by shpran**
**Relates to:** _[ADF-213](https://oat-sa.atlassian.net/browse/ADF-213)_

---

**New roles:**
| # | Role | Abilities | Includes |
| -- | -- | -- | -- |
| AR5 | Asset resource creator | create asset | AR4 (Asset properties editor) |
| AR6 | Asset importer | import single asset | AR5 |
| AR8 | Asset deleter | delete asset | AR3 (Asset previewer) |

---

**Require PR:** _https://github.com/oat-sa/tao-core/pull/3026_
**Companion PR:** _https://github.com/oat-sa/extension-tao-item/pull/494_

### CODE REVIEW

**Comment by shpran** [comment]
@shpran update composer.json versions before merging.

File: model/mapper/MediaSourcePermissionsMapper.php [hunk 1]
```diff
@@ -22,13 +40,21 @@ protected function hasReadAccess(string $uri): bool
...
+        $canDelete = $this->getActionAccessControl()->hasWriteAccess(
+            taoItems_actions_ItemContent::class,
+            'delete'
+        );
+        $canUpload = $this->getActionAccessControl()->hasWriteAccess(
+            taoItems_actions_ItemContent::class,
+            'upload'
+        );
+
+        return parent::hasWriteAccess($uri)&& ($canDelete || $canUpload);
```
Comment (boajer) [inline model/mapper/MediaSourcePermissionsMapper.php:57]:
```suggestion
        return parent::hasWriteAccess($uri) && ($canDelete || $canUpload);
```

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Review by gabrielfs7 — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: model/mapper/MediaSourcePermissionsMapper.php [hunk 2]
```diff
@@ -22,13 +40,21 @@ protected function hasReadAccess(string $uri): bool
             );
     }
 
+    /*
+     *  TODO split permissions to upload and delete assets
```
Comment (gabrielfs7) [inline model/mapper/MediaSourcePermissionsMapper.php:44]: Lets not forget to add this to the new task Dragos created - also align with QA and Dragos.

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Comment by AntonNeviarouski** [comment]
QA verified on http://test-shpran.playground.kitchen.it.taocloud.org:41481/

**Pull request merged**